api/cfnetworking/networking_connection.go

```diff
@@ -0,0 +1,130 @@
+package cfnetworking
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"time"
+
+	"code.cloudfoundry.org/cli/api/cfnetworking/networkerror"
+)
+
+// NetworkingConnection represents a connection to the Cloud Controller
+// server.
+type NetworkingConnection struct {
+	HTTPClient *http.Client
+	UserAgent  string
+}
+
+// Config is for configuring a NetworkingConnection.
+type Config struct {
+	DialTimeout       time.Duration
+	SkipSSLValidation bool
+}
+
+// NewConnection returns a new NetworkingConnection with provided
+// configuration.
+func NewConnection(config Config) *NetworkingConnection {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{
+			InsecureSkipVerify: config.SkipSSLValidation,
+		},
+		Proxy: http.ProxyFromEnvironment,
+		DialContext: (&net.Dialer{
+			KeepAlive: 30 * time.Second,
+			Timeout:   config.DialTimeout,
+		}).DialContext,
+	}
+
+	return &NetworkingConnection{
+		HTTPClient: &http.Client{Transport: tr},
+	}
+}
+
+// Make performs the request and parses the response.
+func (connection *NetworkingConnection) Make(request *Request, passedResponse *Response) error {
+	// In case this function is called from a retry, passedResponse may already
+	// be populated with a previous response. We reset in case there's an HTTP
+	// error and we don't repopulate it in populateResponse.
+	passedResponse.reset()
+
+	response, err := connection.HTTPClient.Do(request.Request)
+	if err != nil {
+		return connection.processRequestErrors(request.Request, err)
+	}
+
+	return connection.populateResponse(response, passedResponse)
+}
+
+func (*NetworkingConnection) processRequestErrors(request *http.Request, err error) error {
+	switch e := err.(type) {
+	case *url.Error:
+		if errors.As(err, &x509.UnknownAuthorityError{}) {
+			return networkerror.UnverifiedServerError{
+				URL: request.URL.String(),
+			}
+		}
+
+		hostnameError := x509.HostnameError{}
+		if errors.As(err, &hostnameError) {
+			return networkerror.SSLValidationHostnameError{
+				Message: hostnameError.Error(),
+			}
+		}
+
+		return networkerror.RequestError{Err: e}
```

Comment:
@ccjaimes - I need help with this chunk of code.  I tried copying the ideas from https://github.com/cloudfoundry/cli/pull/2415 as the failures were the same issue.  But for some reason one test is not getting the same hostname error as in that pr

Reply:
resolving as I reverted (see review comment)